File: android/liveline-core/src/main/kotlin/com/liveline/core/WindowBarTokens.kt
package com.liveline.core

/**
 * Shared design tokens for the interval-bar chrome, so the platform bars stay in
 * lock-step. The bar is grayscale: white on a dark appearance, black on light,
 * at the alphas below. Values mirror web liveline's window-bar styling.
 */
object WindowBarTokens {
    /** Point size for interval labels. */
    const val FONT_SIZE: Double = 11.0

    /** Container/track background alpha. */
    fun trackAlpha(isDark: Boolean): Double = if (isDark) 0.03 else 0.02
    /** Active-button indicator fill alpha. */
    fun indicatorAlpha(isDark: Boolean): Double = if (isDark) 0.06 else 0.035
    /** Active label/icon alpha. */
    fun activeAlpha(isDark: Boolean): Double = if (isDark) 0.7 else 0.55
    /** Inactive label/icon alpha. */
    fun inactiveAlpha(isDark: Boolean): Double = if (isDark) 0.25 else 0.22
}

File: ios/Sources/LivelineKit/Math/WindowBarTokens.swift
import Foundation

/// Shared design tokens for the interval-bar chrome, so the UIKit
/// (`WindowBarView`) and SwiftUI (`WindowBar`) implementations stay in lock-step
/// — change a value here and both bars follow. The bar is grayscale: white on a
/// dark appearance, black on light, at the alphas below. Values mirror web
/// liveline's window-bar styling.
public enum WindowBarTokens {
    /// Point size for interval labels.
    public static let fontSize: Double = 11

    /// Container/track background alpha.
    public static func trackAlpha(isDark: Bool) -> Double { isDark ? 0.03 : 0.02 }
    /// Active-button indicator fill alpha.
    public static func indicatorAlpha(isDark: Bool) -> Double { isDark ? 0.06 : 0.035 }
    /// Active label/icon alpha.
    public static func activeAlpha(isDark: Bool) -> Double { isDark ? 0.7 : 0.55 }
    /// Inactive label/icon alpha.
    public static func inactiveAlpha(isDark: Bool) -> Double { isDark ? 0.25 : 0.22 }
}
